src/commands/force/org/status.ts

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (c) 2021, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import { EOL } from 'os';
+import { flags, FlagsConfig, SfdxCommand } from '@salesforce/command';
+import {
+  Config,
+  Lifecycle,
+  Messages,
+  Aliases,
+  SandboxEvents,
+  StatusEvent,
+  ResultEvent,
+  SandboxProcessObject,
+} from '@salesforce/core';
+import { Duration } from '@salesforce/kit';
+import { SandboxReporter } from '../../../shared/sandboxReporter';
+
+Messages.importMessagesDirectory(__dirname);
+const messages = Messages.loadMessages('@salesforce/plugin-org', 'status');
+
+export class OrgStatusCommand extends SfdxCommand {
+  public static readonly examples = messages.getMessage('examples').split(EOL);
+  public static readonly description = messages.getMessage('description');
+  public static readonly requiresProject = false;
+  public static readonly requiresUsername = true;
+
+  public static readonly flagsConfig: FlagsConfig = {
+    sandboxname: flags.string({
+      char: 'n',
+      description: messages.getMessage('flags.sandboxname'),
+      required: true,
+    }),
+    setdefaultusername: flags.boolean({
+      char: 's',
+      description: messages.getMessage('flags.setdefaultusername'),
+    }),
+    setalias: flags.string({
+      char: 'a',
+      description: messages.getMessage('flags.setalias'),
+    }),
+    wait: flags.minutes({
+      char: 'w',
+      description: messages.getMessage('flags.wait'),
+      min: Duration.minutes(2),
+      default: Duration.minutes(6),
+    }),
+  };
+
+  public async run(): Promise<SandboxProcessObject> {
+    this.logger.debug('Status started with args %s ', this.flags);
+    const lifecycle = Lifecycle.getInstance();
+
+    // eslint-disable-next-line @typescript-eslint/require-await
+    lifecycle.on(SandboxEvents.EVENT_STATUS, async (results: StatusEvent) => {
+      this.ux.log(SandboxReporter.sandboxProgress(results));
+    });
+
+    lifecycle.on(SandboxEvents.EVENT_RESULT, async (results: ResultEvent) => {
+      const resultMsg = `Sandbox ${results.sandboxProcessObj.SandboxName}(${results.sandboxProcessObj.Id}) is ready for use.`;
+      this.ux.log(resultMsg);
+      const { data } = SandboxReporter.logSandboxProcessResult(results);
+      this.ux.styledHeader('Sandbox Org Status');
+      this.ux.table(data, {
+        columns: [
+          { key: 'key', label: 'Name' },
+          { key: 'value', label: 'Value' },
+        ],
+      });
+      if (results.sandboxRes?.authUserName) {
+        if (this.flags.setalias) {
+          const aliases = await Aliases.create(Aliases.getDefaultOptions());
+          const result = await aliases.updateValue(this.flags.setalias, results.sandboxRes.authUserName);
```

Comment:
@WillieRuemmele I made this changes recently because the original source from `toolbel` was not working, can you please test the `-s` and `-a` params they should work ok now